src/rapid_json_serializer.h

```diff
@@ -0,0 +1,50 @@
+#ifndef JINJA2CPP_RAPID_JSON_SERIALIZER_H
+#define JINJA2CPP_RAPID_JSON_SERIALIZER_H
+
+#include "internal_value.h"
+
+#include <rapidjson/document.h>
+#include <rapidjson/rapidjson.h>
+
+#include <memory>
+
+namespace jinja2
+{
+namespace rapidjson_serializer
+{
+
+class ValueWrapper
+{
+    friend class DocumentWrapper;
+
+public:
+    ValueWrapper(ValueWrapper&&) = default;
+    ValueWrapper& operator=(ValueWrapper&&) = default;
+
+    std::string AsString(uint8_t indent = 0) const;
+
+private:
+    ValueWrapper(rapidjson::Value&& value, std::shared_ptr<rapidjson::Document> document);
+
+    rapidjson::Value m_value;
+    std::shared_ptr<rapidjson::Document> m_document;
```

Comment:
Why not unique_ptr?

Reply:
If I understand that correctly from the code, the rapidjson::Document owns the allocator that is used by values. So the lifetime of the Values created with this allocator shouldn't exceed the lifetime of the Document.